### PR TITLE
Fix loan item replace

### PR DESCRIPTION
**Pull request by ntarocco**

### CODE REVIEW

**Comment by ntarocco** [comment]
@FlorianCassayre @KonstantinaStoikou it would be great if you give a careful review to this one since I touched the anonymization module, mainly to make the Anonymous patron class configurable and overridable in CDS-ILS.

File: invenio_app_ils/circulation/mail/messages.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def __init__(self, loan, action, message_ctx, **kwargs):
             )
 
         name = self.get_template_name(action)
-        super(LoanMessage, self).__init__(
```
Comment (FlorianCassayre) [inline invenio_app_ils/circulation/mail/messages.py:48]: :+1: I just learned it was equivalent

**Review by FlorianCassayre — COMMENTED** [review]
Great refactoring, looks good to me!

File: invenio_app_ils/anonymization.py [hunk 2]
```diff
@@ -82,96 +73,103 @@ def anonymize_patron_data(patron_pid, force=False):
...
             params.update(
                 dict(
-                    cancel_reason="Loan request cancelled to anonymize user.",
+                    cancel_reason="Loan request cancelled because of user "
+                    "deletion.",
                     transaction_user_pid=str(SystemAgent.id),
                 )
             )
-            current_circulation.circulation.trigger(
+            loan = current_circulation.circulation.trigger(
```
[the remark this answers is not in the document]
Comment (ntarocco) [reply invenio_app_ils/anonymization.py:116]: Yes it is! It is just my brain that goes automatically in direction of "functional" programming! :) 
Mainly it is that if we change the implementation of the `trigger`, and for some reason `loan` becomes immutable, we don't have to change usage of it!

File: invenio_app_ils/anonymization.py [hunk 3]
```diff
@@ -82,96 +73,103 @@ def anonymize_patron_data(patron_pid, force=False):
...
         DocumentRequestSearch().search_by_patron_pid(patron_pid).scan()
     )
 
+    DocumentRequest = current_app_ils.document_request_record_cls
     for hit in patron_document_requests:
         document_request = DocumentRequest.get_record_by_pid(hit.pid)
         if document_request["state"] == "PENDING":
             document_request["state"] = "REJECTED"
             document_request["reject_reason"] = "USER_CANCEL"
+        print(document_request)
```
[the remark this answers is not in the document]
Comment (ntarocco) [reply invenio_app_ils/anonymization.py]: I will remove the `print(` ;)

File: invenio_app_ils/patrons/api.py [hunk 4]
```diff
@@ -118,28 +119,51 @@ def __init__(self):
         self.location_pid = ""
 
 
+class AnonymousPatron(Patron):
```
Comment (kpsherva) [inline invenio_app_ils/patrons/api.py:122]: Nice! I will adapt it on cds-ils too

**Review by kpsherva — APPROVED** [review]

**Pull request merged**